src/Symfony/Bridge/Doctrine/ContainerAwareEventManager.php

```diff
@@ -189,19 +192,27 @@ public function removeEventSubscriber(EventSubscriber $subscriber): void
 
     private function initializeListeners(string $eventName)
     {
-        $this->initialized[$eventName] = true;
+        if (isset($this->initializing[$eventName])) {
+            throw new EventSubscriberCircularException(sprintf('Circular dependency detected during initialization of listeners for event "%s".', $eventName));
+        }
+        $this->initializing[$eventName] = true;
         foreach ($this->listeners[$eventName] as $hash => $listener) {
             if (\is_string($listener)) {
                 $this->listeners[$eventName][$hash] = $listener = $this->container->get($listener);
 
                 $this->methods[$eventName][$hash] = $this->getMethod($listener, $eventName);
             }
         }
+        $this->initialized[$eventName] = true;
+        unset($this->initializing[$eventName]);
     }
 
     private function initializeSubscribers()
     {
-        $this->initializedSubscribers = true;
+        if ($this->initializingSubscribers) {
+            throw new EventSubscriberCircularException('Circular dependency detected during initialization of event subscribers.');
```

Comment:
Could there be a way to know which one triggers the loop?

Reply:
I can catch the `EventSubscriberCircularException` in the loop to rethrow an other exception with more context, but I didn't want to make code more complex for an info already provided by the stack trace.
![image](https://user-images.githubusercontent.com/400034/212278148-e3bd3049-6841-4dc1-a205-5133a3ac38c5.png)

Reply:
@symfony/mergers Do you think that context details should be added to the exception message, when this information can be found in the stacktrace?

That is, add a try-catch to throw a new exception with more information, pushing the original exception as `$previousException`.

Reply:
I added more context to the message.

For a subscriber:
![image](https://user-images.githubusercontent.com/400034/215296363-df9ece50-5d78-4a72-85bd-9bb7bf0b2579.png)

For a listener:
![image](https://user-images.githubusercontent.com/400034/215296601-1736629b-ab29-4497-963b-2bb658240fb5.png)